powerflex/sdc_volumes_mapping_resource.go

```diff
@@ -681,3 +675,41 @@ func (r *sdcVolumeMappingResource) ImportState(ctx context.Context, req resource
 	// Retrieve import ID and save to id attribute
 	resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
 }
+
+// VerifyVolumes checks if volume exists or not before mapping
+func (r *sdcVolumeMappingResource) VerifyVolumes(ctx context.Context, plan *sdcVolumeMappingResourceModel) (diags diag.Diagnostics) {
+	volList := []sdcVolumeModel{}
+	diag := plan.VolumeList.ElementsAs(ctx, &volList, true)
+	diags.Append(diag...)
+
+	for index, vol := range volList {
+		// Populate volume name in the plan if volume ID is provided in the config
+		if !vol.VolumeID.IsUnknown() {
+			volume, err := r.client.GetVolume("", vol.VolumeID.ValueString(), "", "", false)
+			if err != nil {
+				diags.AddError(
+					"Error getting volume with ID: ",
+					"unexpected error: "+err.Error(),
+				)
+				return
+			}
+			volList[index].VolumeName = types.StringValue(volume[0].Name)
+		} else if !vol.VolumeName.IsUnknown() {
+			volume, err := r.client.GetVolume("", "", "", vol.VolumeName.ValueString(), false)
+			if err != nil {
+				diags.AddError(
+					"Error getting volume with name: ",
+					"unexpected error: "+err.Error(),
+				)
+				return
+			}
+			volList[index].VolumeID = types.StringValue(volume[0].ID)
+		}
```

Comment:
Within ModifyPlan, we could not throw this error as both unknowns was a harmless possibility, but here we should error out.
```suggestion
		} else {
			diags.AddError(
				"Both name and id are unknown for one of the volumes to be mapped.", ""
			)
		}
```

Reply:
All else look good

Reply:
Actually, my bad. Not needed as one of them is mandatory